Tests/SwiftLintFrameworkTests/TestHelpers.swift

```diff
@@ -92,8 +94,9 @@ private extension Collection where Element == StyleViolation {
         return map { violation in
             let locationWithoutFile = Location(file: nil, line: violation.location.line,
                                                character: violation.location.character)
-            return StyleViolation(ruleDescription: violation.ruleDescription, severity: violation.severity,
-                                  location: locationWithoutFile, reason: violation.reason)
+            var new = violation
+            new.location = locationWithoutFile
```

Comment:
again we can use `with(location: locationWithoutFile`) extension
this will remove the need of having mutable property ad well as the  some rules disabing
https://github.com/realm/SwiftLint/pull/3040/files#diff-32f83ec6a61cedc75c20b4319015fff9R6